numpy/lib/polynomial.py

```diff
@@ -552,6 +566,8 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
         raise TypeError("expected 1D or 2D array for y")
     if x.shape[0] != y.shape[0]:
         raise TypeError("expected x and y to have same length")
+    if absolute_weights and (w is None):
```

Comment:
I don't see why this is necessary: the rescaling could be done or omitted (as is arguably meaningful) independent of whether weights are present. I'd remove this.

Reply:
Wouldn't omitting the rescaling without specifying weights effectively mean that all points' standard deviations are considered 1, or did I misunderstand that?

Reply:
Yes, for normal distributions that would be the case. But mostly I see no reason to force a user to pass on `w=np.ones(y.shape[0])` when the flag is set. The default is not really no weights, but weight equals 1.

Reply:
In the case of relative weights (scaling), having all weights set to one means that the error on all data points is of equal magnitude. In the new case (absolute weights, no scaling), I'm not sure if this is any sensible default. Why would a data point have the error sigma==1? I guess, that would be just by coincidence or in special cases where you draw from a distribution with known width (like your unit test example)?

Reply:
@wummo - it indeed implies `sigma=1` - which I agree is not necessarily all that meaningful, but I don't see a reason to specifically forbid someone from entering it - it just makes the code longer and more complex for no benefit.

Reply:
OK, here is my last argument: Isn't sigma==1 a detail of the implementation that might (maybe) change? Then, giving no weights is something like "undefined behavior".

Reply:
But it isn't ;-) After all, this is a possibly weighted least-squares fit, not a chi2 one, and the meaning is clear without the weights (the meaning of the covariance admittedly less so, but I don't think one has to hand-hold people that much).

Reply:
In my opinion, the test is useful, but having very little experience with numpy development/guidlines, I followed your hints and removed the check and the corresponding unit test.